### PR TITLE
Rename UpgradeCannotStart for UpgradeBlockedOnCR

### PR DESCRIPTION
**Pull request by rcarrillocruz**
The UpgradeCannotStart is too similar to UpgradeNotStarted, which
could confuse users.
Since the semantics of this state is to signal that an upgrade
cannot start due to being blocked by another CR which needs to complete
before moving on, just rename it to UpgradeBlockedOnCR.

### CODE REVIEW

File: controllers/clustergroupupgrade_controller.go [hunk 1]
```diff
@@ -135,7 +135,7 @@ func (r *ClusterGroupUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.
 		if readyCondition.Reason == "PrecachingRequired" {
 			requeueAfter := 5 * time.Minute
 			nextReconcile = ctrl.Result{RequeueAfter: requeueAfter}
-		} else if readyCondition.Reason == "UpgradeNotStarted" || readyCondition.Reason == "UpgradeCannotStart" {
+		} else if readyCondition.Reason == "UpgradeNotStarted" || readyCondition.Reason == "UpgradeBlockedOnCR" {
```
Comment (Missxiaoguo) [inline controllers/clustergroupupgrade_controller.go:138]: I think we should create a constants variable for it.

File: controllers/clustergroupupgrade_controller.go [hunk 2]
```diff
@@ -222,7 +222,7 @@ func (r *ClusterGroupUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.
 				meta.SetStatusCondition(&clusterGroupUpgrade.Status.Conditions, metav1.Condition{
 					Type:    "Ready",
 					Status:  metav1.ConditionFalse,
-					Reason:  "UpgradeCannotStart",
+					Reason:  "UpgradeBlockedOnCR",
```
Comment (Missxiaoguo) [inline controllers/clustergroupupgrade_controller.go:225]: UpgradeBlockedOnCR is not accurate for this case - cannot start due to missing policies.

Probably something like "UpgradeNotReadyToStart"?

Comment (rcarrillocruz) [reply controllers/clustergroupupgrade_controller.go:225]: Huh, yes, I did a find/replace.
So this means this particular UpgradeCannotStart has had a dual meaning all this time.
I need to think what's the best transition here.

File: controllers/clustergroupupgrade_controller.go (hunk 1 above)
Comment (rcarrillocruz) [reply controllers/clustergroupupgrade_controller.go:138]: Sure, I can do on a separate PR.

**Comment by irinamihai** [comment]
@rcarrillocruz, KUTTL tests need to be updated. Sorry...

**Pull request closed without merging**